### ISSUE TITLE
nearest-neighbor Tc, ever vs. never generated

### ISSUE THREAD

**Issue by skinnider** [skinniderlab/CLM#217]
writing out the points that @vineetbansal and I talked about earlier this afternoon... 
the idea is, for each molecule in the _test_ set, to record the Tc between this molecule and its nearest-neighbors in the _training_ set. 
These Tc's are then compared for test set molecules that were generated (i.e., rank is not NA/Inf) vs. those never generated by the CLM (i.e. rank NA/Inf).
In the context of cross-validation, the test set is a single fold (e.g. fold 10) and the training set is the remaining folds (e.g. folds 1-9).
Therefore, the `write_nn_tc.py` script should be run with two different files as input, which I believe is not currently the case.

These analyses are depicted in Fig. 4 of the DarkNPS paper:
<img width="400" alt="image" src="https://github.com/skinniderlab/CLM/assets/5023790/df219951-053e-4a7f-966e-46cdf1e0678b">
> **Fig. 4 | Automated structure elucidation of unidentified NPSs.** a, Proportion of molecules within the set of 194 NPSs added to the HighResNPS database between October 2020 and April 2021 that appeared at least once within a sample of one billion SMILES strings from the generative model. b, Tanimoto coefficients between held-out NPSs and their nearest neighbour in the training set, for molecules in the held-out set that were generated at least once versus molecules in the held-out set that were never generated.
